src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleObject.cs

```diff
@@ -22,8 +24,30 @@ private static void HandleStartObject(JsonSerializerOptions options, ref ReadSta
                 // An array of objects either on the current property or on a list
                 Type objType = state.Current.GetElementType();
                 state.Push();
+
                 state.Current.JsonClassInfo = options.GetOrAddClass(objType);
             }
+            else if (state.Current.IsDictionary())
+            {
+                JsonPropertyInfo jsonPropertyInfo = state.Current.JsonPropertyInfo;
+
+                bool skip = jsonPropertyInfo != null && !jsonPropertyInfo.ShouldDeserialize;
```

Comment:
Can you add a comment explaining this condition, especially since we are OR'ing it with something else right after?